### DISCUSSION TITLE
All dimensions in the style are enlarged

### DISCUSSION THREAD

**Discussion by sshallow** [tailwindlabs/tailwindcss#13584] [Help]
I built a Chrome extension and it ran well, but I found that when my content page was injected into the current page, the styles were not displayed correctly. To be precise, all style sizes were enlarged. I used tailwindcss, and the color The category, the italic one looks correct, but the size category is not normal. For example, what I set is: class="flex items-center p-5 py-4 text-sm text-gray-500" but the performance is paddingx is 125, paddingy100, the font size is also enlarged, the setting is 14px, but Actual 87.5px

![1](https://cdn.discordapp.com/attachments/946290204904390690/1233231223103422505/image.png?ex=662c5793&is=662b0613&hm=65c73df0546ed014243b5a1de079f1a38197cc157fa09b2c9a00e27fdf998570&
)
![2](https://cdn.discordapp.com/attachments/946290204904390690/1233231223510536192/image.png?ex=662c5794&is=662b0614&hm=4d9abacc15a84361067371f11b05701440d297462afe4b3a3efb5900240daead&)

**Comment by wongjn** [comment]
By default, many of Tailwind's length values are expressed in [`rem` units](https://developer.mozilla.org/en-US/docs/Web/CSS/length#rem) which "represents the [`font-size`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size) of the root element (typically [`<html>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/html))."

Thus, it seems like the page that your Chrome extension runs on has modified the `font-size` of its `<html>` to be larger than the from the default `16px`:

| Class name | `rem` size | Typical 16px `<html>` font-size | Your web page |
| --- | --- | --- | --- |
| `p-5` | `1.25rem` | `20px`  | `125px` |
| `py-4` | `1rem` | `16px` | `100px` |
| `text-sm` | `0.875rem` | `14px`  | `87.5rem` |

We can see from the `py-4` class, that in the web page, it has set the `font-size` of the `<html>` page to `100px`.

To work-around this, you could consider reworking the values in the Tailwind configuration to use pixel values:

```js
const defaultTheme = require("tailwindcss/defaultTheme");

const remToPx = (value) =>
  value.endsWith("rem") ? `${parseFloat(value.slice(0, -3)) * 16}px` : value;

/** @type {import('tailwindcss').Config} */
export default {
  // …
  theme: {
    extend: {
      spacing: Object.fromEntries(
        Object.entries(defaultTheme.spacing).map(([key, value]) => [
          key,
          remToPx(value),
        ]),
      ),
      fontSize: Object.fromEntries(
        Object.entries(defaultTheme.fontSize).map(([key, value]) => [
          key,
          [
            remToPx(value[0]),
            { ...value[1], lineHeight: remToPx(value[1].lineHeight) },
          ],
        ]),
      ),
    },
  },
};
```

There are some other value sets that use `rem` units. If you need to, follow the example above to set them to pixel values as well.

**Reply by sshallow** [reply]
tks, i found this solution

`npm install @thedutchcoder/postcss-rem-to-px`

`postcss.config.js`
```
module.exports = {
  plugins: {
    tailwindcss: {},
   '@thedutchcoder/postcss-rem-to-px': {}, // you can add option like the base font size
    autoprefixer: {},
  },
};
```

**Reply by iplanwebsites** [reply]
Thanks! For injected chrome extension that's the only fix I found to use tailwind classes.
Might be worth rolling in a helper or adding it to the docs.
